tests/unit/test_local_data_service.py

```diff
@@ -47,7 +47,7 @@ def test_get_dataset():
     mock_cache = MagicMock()
     mock_cache.get.return_value = None
     data_service = LocalDataService.get_instance(
-        config=ConfigService(), cache_service=mock_cache
+        config=ConfigService(), cache_service=mock_cache, data_format="xpt"
```

Comment:
The problem we are seeing in pytest here...
I couldn't figure out how to get the pytest debugger working, but I understand why this test is failing.
In this line, we are assuming that we are getting a new instance of the LocalDataService and that instance uses the data_format=xpt. However, if we run all the tests, this is not the case because LocalDataService uses the singleton pattern and the first time the instance is created in previous tests, it must not be using the xpt data_format.

I'm not sure if a global data_format should be tied to the LocalDataService or not. Will we want to be able to have mixed data formats? If it should be tied to the LDS, then the test can be fixed by adding a mock for retrieving the data format. If it should not be tied to the LDS, then this can be fixed by changing how data_format is represented within the LDS. 

I think maybe we should tie the data_format to the individual datasets instead. This will come naturally if we use the file extensions instead of a cli param anyway. But maybe we should discuss with the team whether or not we should allow mixed data formats.

Reply:
That sounds like a new PBI, couldn't we get around it by creating a new instance of the LocalDataService in the tests manually instead of calling get instance?

Reply:
I thought it was relevant to this PBI since "data_format" being passed around the LDS was introduced in this PR.
And yes, creating a new LDS instance would be much simpler than creating a mock.

Reply:
Ahh ok, my mistake. I had not fully reviewed this PR. I think it makes sense to resolve this issue in this PR and warrants a discussion with the team

Reply:
It's not uncommon to have mixed data formats in a folder but those are usually the SAS7BDAT and their XPT corresponding files. I can't tell if on top of that people will also maintain JSON files or if they'll ditch entirely the XPTs. I would assume they'll keep both for the transtion period from XPT to JSON file formats for regulatory purposes.

Reply:
I think it is best practice to keep json and xpt files in different folders (I would do so if I were still using xpt), but people do not always follow best practices.

Reply:
As agreed with Nick De Donder, we should throw an error if the dataset path contains both XPT and JSON files. I'll update accordingly and assess if data_format parameter is still needed.